ARCdigi/src/ARCdigitizer.cpp

```diff
@@ -0,0 +1,87 @@
+#include "ARCdigitizer.h"
+
+// EDM4HEP
+#include "edm4hep/Vector3d.h"
+
+// DD4HEP
+#include "DDRec/CellIDPositionConverter.h"
+
+// ROOT
+#include "TRandom.h"
+
+// STL
+#include <unordered_map>
+#include <utility>
+
+DECLARE_COMPONENT(ARCdigitizer)
+
+ARCdigitizer::ARCdigitizer(const std::string& aName, ISvcLocator* aSvcLoc) : GaudiAlgorithm(aName, aSvcLoc) {
+  declareProperty("inputSimHits", m_input_sim_hits, "Input sim tracker hit collection name");
+  declareProperty("outputDigiHits", m_output_digi_hits, "Output digitized tracker hit collection name");
+  declareProperty("detectorCompact", m_det_compact, "Path to detector compact");
+  declareProperty("flatSiPMEfficiency", m_flat_SiPM_effi, "Flat value for SiPM efficiency (<0 := disabled)");
+  declareProperty("applySiPMEffiToDigiHits", m_apply_SiPM_effi_to_digi,
+                  "Apply the SiPM efficiency to digitized hits instead of simulated hits");
+}
+
+ARCdigitizer::~ARCdigitizer() {}
+
+StatusCode ARCdigitizer::initialize() {
+  m_detector = dd4hep::Detector::make_unique(this->name() + "_detector");
+  if (m_det_compact.value().empty()) {
```

Comment:
To be removed, add a check that the geoService is available instead

Reply:
Done!